dv/cosim/spike_cosim.cc

```diff
@@ -200,7 +200,10 @@ bool SpikeCosim::step(uint32_t write_reg, uint32_t write_reg_data,
 
   if (processor->get_state()->last_inst_pc == PC_INVALID) {
     if (!(processor->get_state()->mcause->read() & 0x80000000) ||
-        processor->get_state()->debug_mode) { // (Async-Traps are disabled in debug mode)
+        (processor->get_state()->mcause->read() ==
```

Comment:
Internal NMIs should be handled as an async trap here, had you encountered difficulties using internal NMI with the sync trap here?

Reply:
I tested removing it now, it didn't changed the pass rate. Possibly it was some other problem that lead me to believe this was the correct behaviour. I'll change it to behave just as another NMI now.